spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -331,9 +331,13 @@ protected void actualStart() throws Exception {
 					this.lastAlertAt = now;
 				}
 			}
-			this.consumers.stream()
-					.filter(c -> !c.getChannel().isOpen())
-					.collect(Collectors.toList()) // needed to avoid ConcurrentModificationException in cancelConsumer()
+			final List<SimpleConsumer> consumersToCancel;
+			synchronized (this.consumersMonitor) {
+				consumersToCancel = this.consumers.stream()
```

Comment:
Synchronization is needed to be sure that this operation succeeds even if other threads try to modify it. Please give a look at the class [documentation](https://docs.oracle.com/javase/7/docs/api/java/util/LinkedList.html).